### PR TITLE
Out with the old, in with the new

### PR DESCRIPTION
**Pull request by mcmire**
This commit is long overdue. The main change is to upgrade shoulda's
hard dependency on shoulda-matchers from 2.x to 3.x. Since this by
itself is a pretty consequential change, though, I also took the
opportunity to make some more at the same time that I think are
reasonable:
- Remove support for Rails 3, upgrading dependency to 4, modifying
  Appraisals file and fixing Cucumber specs to match.
- Remove support for RSpec, along with related Cucumber specs. People
  who use RSpec won't be using the `shoulda` gem -- they'll be using
  `shoulda-matchers` instead. There's already a good collection of
  integration specs there, so we don't need the tests here.
- Remove support for Ruby 1.9, along with the entry in .travis.yml.

Due to the backward-incompatible nature of these changes, my thought is
to release a version 4 of shoulda that includes them.

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -12,3 +15,42 @@
...
+      :minitest_5
+    else
+      :minitest_4
+    end
+  end
+end
+
+module RailsHelpers
+  def rails_version
+    Gem::Version.new(Bundler.definition.specs['rails'][0].version)
```
Comment (mcmire) [inline features/support/env.rb]: I don't think a type of quote is being used consistently in this project, although single quotes seems to be used more often. I'm going to ignore these warnings.

**Comment by mcmire** [comment]
@rmm5t Since you have done a lot of work on this for a while, what do you think about this change?

**Comment by rmm5t** [comment]
@mcmire Fine by me. I only rely on shoulda-matchers directly anymore. I just have one legacy project that still has a direct shoulda dependency, but upon quick glance, these dependency changes look sane to me.

**Comment by samuelpismel** [comment]
No merge?

**Comment by romdim** [comment]
@rmm5t / @mcmire Could you please merge? Thanks!

**Comment by rmm5t** [comment]
@romdim It's been a while since this branch's build ran green. I'm happy to merge, but I don't have the time this week to debug in the event that new versions of dependencies turn the build red. @mcmire Do you have time to merge and release a new gem version?

**Comment by mcmire** [comment]
@rmm5t Sure. I'll take a look at this some time this week.

**Pull request merged**